src/instance.rs

```diff
@@ -995,6 +995,7 @@ impl PyObject {
     ///
     /// This can cast only to native Python types, not types implemented in Rust. For a more
     /// flexible alternative, see [`Py::extract`](struct.Py.html#method.extract).
+    #[inline]
     pub fn downcast<'p, T>(&'p self, py: Python<'p>) -> Result<&T, PyDowncastError<'_>>
```

Comment:
`#[inline]` here does nothing, because generic, functions are already implicitly inlineable (same for every other inline in this PR)

Reply:
Actually I think there is still some benefit - it's true that llvm may always choose to inline generics but adding `#[inline]` does add `inline` hint to the llvm code so may encourage llvm to inline when it might have decided not to.